File: Ywy/JavaScript/level1/test/ColatzGuess.test.js
import ColatzGuess from '../ColatzGuess';

test("6 is 8", () => {
    expect(new ColatzGuess().solution(6)).toBe(8);
});

test("16 is 4", () => {
    expect(new ColatzGuess().solution(16)).toBe(4);
});

test("626331 is -1", () => {
    expect(new ColatzGuess().solution(626331)).toBe(-1);
});

File: Ywy/JAVA/src/level1/test/cases/ColatzGuessTest.java
package level1.test.cases;

import static org.junit.Assert.assertEquals;

import org.junit.Before;
import org.junit.Test;

import level1.ColatzGuess;

public class ColatzGuessTest {
	private int[] ns;
	private int[] answers;
	private ColatzGuess colatzGuess;
	
	@Before
	public void setUp() {
		ns = new int[] {
			6,
			16,
			626331
		};
		answers = new int[] {
			8,
			4,
			-1
		};
		colatzGuess = new ColatzGuess();
	}
	
	@Test
	public void test() {
		int index = 0;
		int range = answers.length;
		do {
			assertEquals(colatzGuess.solution(ns[index]), answers[index]);
		} while(++index < range);
	}
}
